unic-locale-impl/src/extensions/private.rs

```diff
@@ -16,16 +16,91 @@ fn parse_value(t: &str) -> Result<TinyStr8, ParserError> {
 }
 
 impl PrivateExtensionList {
+    /// Returns `true` if there are no tags in the PrivateExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-x-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.private.is_empty(), false);
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
 
+    /// Returns `true` if tag is included in the `PrivateExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-x-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.private.get_tag("foo")
+    ///                .expect("Getting tag failed."),
+    ///            true);
+    /// ```
+    pub fn get_tag(&self, tag: &str) -> Result<bool, LocaleError> {
+        Ok(self.0.contains(&parse_value(tag)?))
+    }
+
+    /// Adds a tag to the `PrivateExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// lo.extensions.private.add_tag("foo")
+    ///     .expect("Adding tag failed.");
+    ///
+    /// assert_eq!(lo.to_string(), "en-US-x-foo");
+    /// ```
     pub fn add_tag(&mut self, tag: &str) -> Result<(), LocaleError> {
         self.0.push(parse_value(tag)?);
         self.0.sort();
         Ok(())
     }
 
+    /// Removes a tag from the `PrivateExtensionList`.
+    ///
+    /// Returns `true` if tag was included in the `PrivateExtensionList` before
+    /// removal.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-x-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.private.remove_tag("foo")
+    ///                .expect("Removing tag failed."),
+    ///            true);
+    ///
+    /// assert_eq!(lo.to_string(), "en-US");
+    /// ```
+    pub fn remove_tag(&mut self, tag: &str) -> Result<bool, LocaleError> {
+        let value = parse_value(tag)?;
+        match self.0.binary_search(&value) {
+            Ok(idx) => {
+                self.0.remove(idx);
+                Ok(true)
+            },
+            Err(_) => Ok(false)
+        }
+    }
+
```

Comment:
Can you add `clear_tags`?

Reply:
And `get_tags` which returns `impl ExactSizeIterator<Item = &str>` (much like `get_variants` does)

Reply:
Hmm, maybe name it `get_tag_keys` since I think we should return just keys, not values.

Reply:
I'll leave `get_tags` for a separate commit. Other thing is resolved.